Failing integration test for system tests does not show origin failure but system test NPE

## Situation
In case of running `./sdc.sh --integrationtest-systemtest`  the next errors apears:
`SystemTestFrameworkIntTest > even_integration_test_setup_can_be_tested_codescan_source_only_and_gen_example() FAILED
    java.lang.NullPointerException at SystemTestFrameworkIntTest.java:146`
`SystemTestFrameworkIntTest > even_integration_test_setup_can_be_tested__codescan_with_binary_and_sources() FAILED
    java.lang.NullPointerException at SystemTestFrameworkIntTest.java:239`

<details>
  <summary>Details</summary>
even_integration_test_setup_can_be_tested__codescan_with_binary_and_sources()
java.lang.NullPointerException: Cannot invoke "com.mercedesbenz.sechub.api.SecHubReport.getTrafficLight()" because "report" is null
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.handleSecHubAssert(SystemTestRuntimeTestAssertion.java:42)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.handleSecHubAsserts(SystemTestRuntimeTestAssertion.java:33)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.assertTest(SystemTestRuntimeTestAssertion.java:23)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestEngine.assertTest(SystemTestRuntimeTestEngine.java:86)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestEngine.runTest(SystemTestRuntimeTestEngine.java:73)
	at com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntime.runAfterInitialization(SystemTestRuntime.java:122)
	at com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntime.run(SystemTestRuntime.java:73)
	at com.mercedesbenz.sechub.systemtest.SystemTestAPI.runSystemTests(SystemTestAPI.java:65)
	at com.mercedesbenz.sechub.systemtest.SystemTestFrameworkIntTest.even_integration_test_setup_can_be_tested__codescan_with_binary_and_sources(SystemTestFrameworkIntTest.java:239)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:110)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:90)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:85)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
   `
    
even_integration_test_setup_can_be_tested_codescan_source_only_and_gen_example()
java.lang.NullPointerException: Cannot invoke "com.mercedesbenz.sechub.api.SecHubReport.getTrafficLight()" because "report" is null
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.handleSecHubAssert(SystemTestRuntimeTestAssertion.java:42)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.handleSecHubAsserts(SystemTestRuntimeTestAssertion.java:33)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestAssertion.assertTest(SystemTestRuntimeTestAssertion.java:23)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestEngine.assertTest(SystemTestRuntimeTestEngine.java:86)
	at com.mercedesbenz.sechub.systemtest.runtime.testengine.SystemTestRuntimeTestEngine.runTest(SystemTestRuntimeTestEngine.java:73)
	at com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntime.runAfterInitialization(SystemTestRuntime.java:122)
	at com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntime.run(SystemTestRuntime.java:73)
	at com.mercedesbenz.sechub.systemtest.SystemTestAPI.runSystemTests(SystemTestAPI.java:65)
	at com.mercedesbenz.sechub.systemtest.SystemTestFrameworkIntTest.even_integration_test_setup_can_be_tested_codescan_source_only_and_gen_example(SystemTestFrameworkIntTest.java:146)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:110)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:90)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:85)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
</details>

## Wanted
`./sdc.sh --integrationtest-systemtest`  must run without errors.

@de-jcup

@de-jcup @Jeeppler 

I ran the tests again on GitHub Actions. Unfortunately they failed again with the same NPE:
`BUILD FAILED in 51s
[com.mercedesbenz.sechub.systemtest.SystemTestFrameworkIntTest::even_integration_test_setup_can_be_tested_codescan_source_only_and_gen_example()]
[com.mercedesbenz.sechub.systemtest.SystemTestFrameworkIntTest::even_integration_test_setup_can_be_tested__codescan_with_binary_and_sources()]`
[failed tests](https://github.com/zigfridus/sechub/actions/runs/6429764989/job/17460820482#step:10:167)

Looks like the problem is on my side only.
I could try to update system tests by myself, but I need the set up guide of Eclipse for debugging the system tests.

Thanks

--- a/sechub-api-java/src/main/java/com/mercedesbenz/sechub/api/DefaultSecHubClient.java
+++ b/sechub-api-java/src/main/java/com/mercedesbenz/sechub/api/DefaultSecHubClient.java
@@ -439,27 +439,21 @@ public class DefaultSecHubClient extends AbstractSecHubClient {
     private void runOrFail(RunOrFail<ApiException> failable, String failureMessage) throws SecHubClientException {
         try {
             failable.runOrFail();
-        } catch (ApiException e) {
+        } catch (Exception e) {
+            throw createClientException(failureMessage, e);
+        }
+    }
+
+    private <T> T runOrFail(Callable<T> callable, String failureMessage) throws SecHubClientException {
+        try {
+            return callable.call();
+        } catch (Exception e) {
             throw createClientException(failureMessage, e);
         }
     }
 
     private SecHubClientException createClientException(String message, Exception cause) throws SecHubClientException {
         return new SecHubClientException(message + " - " + cause.getMessage(), cause);
-    }
-
-    private <T> T runOrFail(Callable<T> callable, String failureMessage) throws SecHubClientException {
-        try {
-            return callable.call();
-        } catch (ApiException e) {
-            throw createClientException(failureMessage, e);
-        } catch (Exception e) {
-            if (e instanceof RuntimeException) {
-                RuntimeException re = (RuntimeException) e;
-                throw re;
-            }
-            throw new IllegalStateException("Unhandled exception - should not happen", e);
-        }
     }
 
 }

--- a/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/SystemTestRuntimeContext.java
+++ b/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/SystemTestRuntimeContext.java
@@ -56,7 +56,7 @@ public class SystemTestRuntimeContext {
     Path additionalResourcesRoot;
 
     private SystemTestConfiguration configuration;
-    private SystemTestRunResult currentResult;
+    private SystemTestRunResult currentResult = new SystemTestRunResult("no-test-fallback-" + System.nanoTime());
     private Set<SystemTestRunResult> results = new LinkedHashSet<>();
     private SystemTestRuntimeMetaData runtimeMetaData = new SystemTestRuntimeMetaData();
     private Map<PDSSolutionDefinition, PDSSolutionRuntimeData> pdsSolutionRuntimeDataMap = new LinkedHashMap<>();

--- a/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/CurrentTestVariableCalculatorFactory.java
+++ b/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/CurrentTestVariableCalculatorFactory.java
@@ -1,0 +1,9 @@
+package com.mercedesbenz.sechub.systemtest.runtime.testengine;
+
+import com.mercedesbenz.sechub.systemtest.config.TestDefinition;
+import com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntimeContext;
+
+interface CurrentTestVariableCalculatorFactory {
+
+    CurrentTestVariableCalculator create(TestDefinition test, SystemTestRuntimeContext context);
+}

--- a/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestAssertion.java
+++ b/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestAssertion.java
@@ -5,6 +5,7 @@ import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
 import com.mercedesbenz.sechub.api.SecHubReport;
@@ -24,17 +25,30 @@ public class SystemTestRuntimeTestAssertion {
     }
 
     private void handleSecHubAsserts(TestAssertDefinition assertDefinition, TestEngineTestContext testContext) {
+        if (testContext.hasFailed()) {
+            // already marked as failed - e.g. by other test asserts
+            return;
+        }
+
         if (!testContext.isSecHubTest()) {
             return;
         }
-        if (assertDefinition.getSechubResult().isEmpty()) {
+
+        Optional<AssertSechubResultDefinition> sechubResult = assertDefinition.getSechubResult();
+        if (sechubResult.isEmpty()) {
             return;
         }
-        handleSecHubAssert(assertDefinition.getSechubResult().get(), testContext);
+
+        handleSecHubAssert(sechubResult.get(), testContext);
     }
 
     private void handleSecHubAssert(AssertSechubResultDefinition sechubResultAssert, TestEngineTestContext testContext) {
         SecHubReport report = testContext.getSecHubRunData().getReport();
+        if (report == null) {
+            testContext.markAsFailed("Sechub report not available",
+                    "This should not happen. Please look at the unit test console log and inside SecHub server log for details");
+            return;
+        }
         String reportAsJson = JSONConverter.get().toJSON(report, true);
 
         if (sechubResultAssert.getHasTrafficLight().isPresent()) {

--- a/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngine.java
+++ b/sechub-systemtest/src/main/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngine.java
@@ -15,7 +15,6 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import com.mercedesbenz.sechub.api.SecHubClient;
-import com.mercedesbenz.sechub.api.SecHubClientException;
 import com.mercedesbenz.sechub.api.SecHubReport;
 import com.mercedesbenz.sechub.commons.model.JSONConverter;
 import com.mercedesbenz.sechub.commons.model.SecHubConfigurationModel;
@@ -55,6 +54,7 @@ public class SystemTestRuntimeTestEngine {
 
     RunSecHubJobDefinitionTransformer runSecHubJobTransformer = new RunSecHubJobDefinitionTransformer();
     SystemTestRuntimeTestAssertion testAssertion = new SystemTestRuntimeTestAssertion();
+    CurrentTestVariableCalculatorFactory currentTestVariableCalculatorFactory = new DefaultCurrentTestVariableCalculatorFactory();
 
     public SystemTestRuntimeTestEngine(ExecutionSupport execSupport) {
         this.execSupport = execSupport;
@@ -63,10 +63,7 @@ public class SystemTestRuntimeTestEngine {
     public void runTest(TestDefinition test, SystemTestRuntimeContext runtimeContext) {
         TestEngineTestContext testContext = createTestContext(test, runtimeContext);
 
-        if (!prepareTest(testContext)) {
-            /* preparation failed */
-            return;
-        }
+        prepareTest(testContext);
 
         executeTest(testContext);
 
@@ -81,29 +78,38 @@ public class SystemTestRuntimeTestEngine {
     }
 
     private void assertTest(TestEngineTestContext testContext) {
+        if (testContext.hasFailed()) {
+            // already marked as failed
+            return;
+        }
         List<TestAssertDefinition> asserts = testContext.getTest().getAssert();
         for (TestAssertDefinition assertDef : asserts) {
             testAssertion.assertTest(assertDef, testContext);
         }
     }
 
-    private boolean prepareTest(TestEngineTestContext testContext) {
-        boolean prepared = false;
+    private void prepareTest(TestEngineTestContext testContext) {
+        if (testContext.hasFailed()) {
+            // already marked as failed
+            return;
+        }
         try {
             executePreparationSteps("Prepare", testContext);
-            prepared = true;
-        } catch (SystemTestScriptExecutionException e) {
+        } catch (Exception e) {
             testContext.markAsFailed("Was not able to prepare test", e);
         }
-        return prepared;
     }
 
     private void executeTest(TestEngineTestContext testContext) {
+        if (testContext.hasFailed()) {
+            // already marked as failed
+            return;
+        }
         /* mark current test - fail if not supported */
         if (testContext.isSecHubTest()) {
             try {
                 launchSecHubJob(testContext);
-            } catch (SecHubClientException e) {
+            } catch (Exception e) {
                 testContext.markAsFailed("Was not able to launch SecHub job", e);
             }
         } else {
@@ -113,7 +119,7 @@ public class SystemTestRuntimeTestEngine {
         }
     }
 
-    private void launchSecHubJob(TestEngineTestContext testContext) throws SecHubClientException {
+    private void launchSecHubJob(TestEngineTestContext testContext) throws Exception {
         SecHubClient clientForScheduling = null;
 
         SystemTestRuntimeContext runtimeContext = testContext.getRuntimeContext();
@@ -228,8 +234,8 @@ public class SystemTestRuntimeTestEngine {
         String from = copyDirectoriesDefinition.getFrom();
         String to = copyDirectoriesDefinition.getTo();
 
-        from = testContext.dynamicVariableGenerator.calculateValue(from);
-        to = testContext.dynamicVariableGenerator.calculateValue(to);
+        from = testContext.currentTestVariableCalculator.calculateValue(from);
+        to = testContext.currentTestVariableCalculator.calculateValue(to);
 
         try {
             Path source = Paths.get(from).toRealPath();
@@ -251,7 +257,8 @@ public class SystemTestRuntimeTestEngine {
 
     private void executePreparationScript(TestEngineTestContext testContext, ScriptDefinition scriptDefinition) throws SystemTestScriptExecutionException {
 
-        ProcessContainer processContainer = execSupport.execute(scriptDefinition, testContext.getDynamicVariableGenerator(), SystemTestExecutionState.PREPARE);
+        ProcessContainer processContainer = execSupport.execute(scriptDefinition, testContext.getCurrentTestVariableCalculator(),
+                SystemTestExecutionState.PREPARE);
 
         long startTime = System.currentTimeMillis();
         long diffTime = startTime;
@@ -307,25 +314,34 @@ public class SystemTestRuntimeTestEngine {
 
     }
 
+    class DefaultCurrentTestVariableCalculatorFactory implements CurrentTestVariableCalculatorFactory {
+
+        @Override
+        public CurrentTestVariableCalculator create(TestDefinition test, SystemTestRuntimeContext runtimeContext) {
+            return new CurrentTestVariableCalculator(test, runtimeContext);
+        }
+
+    }
+
     class TestEngineTestContext {
 
         private final SystemTestRuntimeTestEngine systemTestRuntimeTestEngine;
         private SecHubRunData secHubRunData;
         SystemTestRuntimeContext runtimeContext;
         TestDefinition test;
-        private CurrentTestVariableCalculator dynamicVariableGenerator;
+        private CurrentTestVariableCalculator currentTestVariableCalculator;
 
         TestEngineTestContext(SystemTestRuntimeTestEngine systemTestRuntimeTestEngine, TestDefinition test, SystemTestRuntimeContext runtimeContext) {
             this.systemTestRuntimeTestEngine = systemTestRuntimeTestEngine;
             this.test = test;
             this.runtimeContext = runtimeContext;
-            this.dynamicVariableGenerator = new CurrentTestVariableCalculator(test, runtimeContext);
+            this.currentTestVariableCalculator = currentTestVariableCalculatorFactory.create(test, runtimeContext);
 
             appendSecHubRunData();
         }
 
-        public VariableCalculator getDynamicVariableGenerator() {
-            return dynamicVariableGenerator;
+        public VariableCalculator getCurrentTestVariableCalculator() {
+            return currentTestVariableCalculator;
         }
 
         public TestDefinition getTest() {
@@ -361,7 +377,7 @@ public class SystemTestRuntimeTestEngine {
 
             String configurationAsJson = converter.toJSON(secHubConfiguration);
 
-            String changedConfigurationAsJson = dynamicVariableGenerator.replace(configurationAsJson);
+            String changedConfigurationAsJson = currentTestVariableCalculator.replace(configurationAsJson);
 
             secHubRunData.secHubConfiguration = converter.fromJSON(SecHubConfigurationModel.class, changedConfigurationAsJson);
         }
@@ -383,6 +399,10 @@ public class SystemTestRuntimeTestEngine {
             failure.setDetails(createDetails(hint, e));
 
             runtimeContext.getCurrentResult().setFailure(failure);
+        }
+
+        public boolean hasFailed() {
+            return runtimeContext.getCurrentResult().hasFailed();
         }
 
         private String createDetails(String hint, Exception e) {

--- a/sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/TestRuntimeAccess.java
+++ b/sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/TestRuntimeAccess.java
@@ -1,0 +1,8 @@
+package com.mercedesbenz.sechub.systemtest.runtime;
+
+public class TestRuntimeAccess {
+
+    public static SystemTestRunResult createDummyTestRunResult() {
+        return new SystemTestRunResult("dummy");
+    }
+}

--- a/sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngineTest.java
+++ b/sechub-systemtest/src/test/java/com/mercedesbenz/sechub/systemtest/runtime/testengine/SystemTestRuntimeTestEngineTest.java
@@ -1,0 +1,182 @@
+package com.mercedesbenz.sechub.systemtest.runtime.testengine;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.mercedesbenz.sechub.api.JobStatus;
+import com.mercedesbenz.sechub.api.SecHubClient;
+import com.mercedesbenz.sechub.api.SecHubClientException;
+import com.mercedesbenz.sechub.api.SecHubReport;
+import com.mercedesbenz.sechub.commons.model.JSONConverter;
+import com.mercedesbenz.sechub.commons.model.SecHubConfigurationModel;
+import com.mercedesbenz.sechub.commons.model.TrafficLight;
+import com.mercedesbenz.sechub.commons.model.job.ExecutionResult;
+import com.mercedesbenz.sechub.systemtest.config.AssertSechubResultDefinition;
+import com.mercedesbenz.sechub.systemtest.config.RunSecHubJobDefinition;
+import com.mercedesbenz.sechub.systemtest.config.TestAssertDefinition;
+import com.mercedesbenz.sechub.systemtest.config.TestDefinition;
+import com.mercedesbenz.sechub.systemtest.runtime.LocationSupport;
+import com.mercedesbenz.sechub.systemtest.runtime.SystemTestRunResult;
+import com.mercedesbenz.sechub.systemtest.runtime.SystemTestRuntimeContext;
+import com.mercedesbenz.sechub.systemtest.runtime.TestRuntimeAccess;
+import com.mercedesbenz.sechub.systemtest.runtime.launch.ExecutionSupport;
+
+class SystemTestRuntimeTestEngineTest {
+
+    private static final TrafficLight EXPECTED_TRAFFIC_LIGHT_YELLOW = TrafficLight.YELLOW;
+    private SystemTestRuntimeTestEngine engineToTest;
+    private ExecutionSupport execSupport;
+    private CurrentTestVariableCalculatorFactory currentTestVariableCalculatorFactory;
+    private CurrentTestVariableCalculator calculator;
+    private SystemTestRuntimeContext runtimeContext;
+    private SecHubClient secHubClient;
+    private LocationSupport locationSupport;
+    private SystemTestRunResult currentTestResult;
+    private TestAssertDefinition assertDefinition;
+
+    @BeforeEach
+    void beforeEach() {
+        execSupport = mock(ExecutionSupport.class);
+        calculator = mock(CurrentTestVariableCalculator.class);
+        runtimeContext = mock(SystemTestRuntimeContext.class);
+        currentTestVariableCalculatorFactory = mock(CurrentTestVariableCalculatorFactory.class);
+        secHubClient = mock(SecHubClient.class);
+        locationSupport = mock(LocationSupport.class);
+        currentTestResult = TestRuntimeAccess.createDummyTestRunResult();
+        assertDefinition = mock(TestAssertDefinition.class);
+
+        engineToTest = new SystemTestRuntimeTestEngine(execSupport);
+        engineToTest.currentTestVariableCalculatorFactory = currentTestVariableCalculatorFactory;
+    }
+
+    @Test
+    void sechub_client_without_errors_assertion_done__variant_valid() throws Exception {
+        /* prepare */
+        JobStatus status = new JobStatus();
+        status.setResult(ExecutionResult.OK);
+        when(secHubClient.fetchJobStatus(any(), any())).thenReturn(status);
+        SecHubReport report = new SecHubReport();
+        report.setTrafficLight(EXPECTED_TRAFFIC_LIGHT_YELLOW);
+
+        when(secHubClient.downloadSecHubReportAsJson(any(), any())).thenReturn(report);
+
+        TestDefinition test = configureSecHubLocalRunAndReturnTestDefinition();
+
+        /* execute */
+        engineToTest.runTest(test, runtimeContext);
+
+        /* test */
+        assertFalse(currentTestResult.hasFailed());
+        assertNull(currentTestResult.getFailure());
+
+        // check assert definition interactions where done
+        verify(assertDefinition).getSechubResult();
+
+    }
+
+    @Test
+    void sechub_client_without_errors_assertion_done__variant_assertion_fails_for_trafficlight() throws Exception {
+        /* prepare */
+        JobStatus status = new JobStatus();
+        status.setResult(ExecutionResult.OK);
+        when(secHubClient.fetchJobStatus(any(), any())).thenReturn(status);
+        SecHubReport report = new SecHubReport();
+        report.setTrafficLight(TrafficLight.RED);
+
+        when(secHubClient.downloadSecHubReportAsJson(any(), any())).thenReturn(report);
+
+        TestDefinition test = configureSecHubLocalRunAndReturnTestDefinition();
+
+        /* execute */
+        engineToTest.runTest(test, runtimeContext);
+
+        /* test */
+        assertTrue(currentTestResult.hasFailed());
+        assertEquals("SecHub report not as wanted. Expected was traffic light: YELLOW, but result was: RED", currentTestResult.getFailure().getMessage());
+
+        // check assert definition interactions where done
+        verify(assertDefinition).getSechubResult();
+
+    }
+
+    @Test
+    void sechub_job_creation_fails() throws Exception {
+        /* prepare */
+        when(secHubClient.createJob(any())).thenThrow(new SecHubClientException("no job createable"));
+
+        TestDefinition test = configureSecHubLocalRunAndReturnTestDefinition();
+
+        /* execute */
+        engineToTest.runTest(test, runtimeContext);
+
+        /* test */
+        assertEquals("Was not able to launch SecHub job", currentTestResult.getFailure().getMessage());
+        assertTrue(currentTestResult.getFailure().getDetails().contains("no job createable"));
+        assertTrue(currentTestResult.hasFailed());
+        verifyNoInteractions(assertDefinition);
+    }
+
+    @Test
+    void sechub_download_fails() throws Exception {
+        /* prepare */
+        JobStatus status = new JobStatus();
+        status.setResult(ExecutionResult.OK);
+        when(secHubClient.fetchJobStatus(any(), any())).thenReturn(status);
+        when(secHubClient.downloadSecHubReportAsJson(any(), any())).thenThrow(new NullPointerException());
+        TestDefinition test = configureSecHubLocalRunAndReturnTestDefinition();
+
+        /* execute */
+        engineToTest.runTest(test, runtimeContext);
+
+        /* test */
+        assertEquals("Was not able to launch SecHub job", currentTestResult.getFailure().getMessage());
+        assertTrue(currentTestResult.getFailure().getDetails().contains("java.lang.NullPointerException"));
+        assertTrue(currentTestResult.hasFailed());
+    }
+
+    @Test
+    void sechub_status_fails() throws Exception {
+        /* prepare */
+        when(secHubClient.fetchJobStatus(any(),any())).thenThrow(new SecHubClientException("no status readable"));
+
+        TestDefinition test = configureSecHubLocalRunAndReturnTestDefinition();
+
+        /* execute */
+        engineToTest.runTest(test, runtimeContext);
+
+        /* test */
+        assertEquals("Was not able to launch SecHub job", currentTestResult.getFailure().getMessage());
+        assertTrue(currentTestResult.getFailure().getDetails().contains("no status readable"));
+        assertTrue(currentTestResult.hasFailed());
+    }
+
+    private TestDefinition configureSecHubLocalRunAndReturnTestDefinition() {
+        when(runtimeContext.isLocalRun()).thenReturn(true);
+        when(runtimeContext.getLocalAdminSecHubClient()).thenReturn(secHubClient);
+        when(currentTestVariableCalculatorFactory.create(any(), any())).thenReturn(calculator);
+        when(runtimeContext.getLocationSupport()).thenReturn(locationSupport);
+        when(runtimeContext.getCurrentResult()).thenReturn(currentTestResult);
+
+        SecHubConfigurationModel dummyModel = new SecHubConfigurationModel();
+        when(calculator.replace(any())).thenReturn(JSONConverter.get().toJSON(dummyModel));
+
+
+        AssertSechubResultDefinition sechubResult = new AssertSechubResultDefinition();
+        sechubResult.setHasTrafficLight(Optional.of(EXPECTED_TRAFFIC_LIGHT_YELLOW));
+        when(assertDefinition.getSechubResult()).thenReturn(Optional.of(sechubResult));
+
+        TestDefinition test = new TestDefinition();
+        RunSecHubJobDefinition sechubJobDefinition = new RunSecHubJobDefinition();
+        test.getExecute().setRunSecHubJob(Optional.of(sechubJobDefinition));
+        test.getAssert().add(assertDefinition);
+
+
+        return test;
+    }
+}

